core/src/main/java/org/apache/iceberg/PartitionSpecParser.java

```diff
@@ -66,19 +65,7 @@ public static String toJson(UnboundPartitionSpec spec) {
   }
 
   public static String toJson(UnboundPartitionSpec spec, boolean pretty) {
-    try {
-      StringWriter writer = new StringWriter();
-      JsonGenerator generator = JsonUtil.factory().createGenerator(writer);
-      if (pretty) {
-        generator.useDefaultPrettyPrinter();
-      }
-      toJson(spec, generator);
-      generator.flush();
-      return writer.toString();
-
-    } catch (IOException e) {
-      throw new RuntimeIOException(e);
-    }
+    return JsonUtil.generate(gen -> toJson(spec, gen), pretty);
```

Comment:
Nice!

BTW we have  a bit duplicated code for parsing as well. I was changing it in https://github.com/apache/iceberg/pull/5048.

BTW 2: i'd use `generator` instead of `gen`. That's how it's usually called, so it will be more familiar

Reply:
> i'd use `generator` instead of `gen`.

i see now both are used in the codebase. disregard

Reply:
> BTW we have a bit duplicated code for parsing as well. I was changing it in https://github.com/apache/iceberg/pull/5048.

ah interesting, you're right. Do you want to open maybe a separate PR for removing that part of duplicated code?